### DISCUSSION TITLE
[suggestion] Merge OpenCore and Clover installers

### DISCUSSION THREAD

**Discussion by noyannus** [chris1111/HP-ProbookEliteBook-Package-Creator-Clover#13] [Ideas]
You could merge the Package Creators for Clover and OpenCore into a single project. 

At the beginning of the main script, ask for the desired boot loader, and set the choice as a variable. It can serve as direct variable (as below) and as a flag in if-statements (if there are more differences that I have overlooked).

In pseudo-code:
```
Ask "Which bootloader do you want?" with the options "Clover" and "OpenCore"
Set BL to the return
...
set progress additional description to "\nCreation of Install Media with " & BL & " bootloader complete!"
delay 3
set theFile to ((path to me) as string) & "Contents:Resources:Installer:" & BL & " USB.pkg"
tell application "Finder" to open theFile
...
```

This will save you from maintaining two almost identical projects.

**Comment by chris1111** [comment]
Thanks for the suggestion
I will never mixed OC and Clover
